dex-it/src/test/scala/com/wavesplatform/it/matcher/api/http/rates/UpdateRatesByAssetIdSpec.scala

```diff
@@ -0,0 +1,74 @@
+package com.wavesplatform.it.matcher.api.http.rates
+
+import com.softwaremill.sttp.StatusCodes
+import com.typesafe.config.{Config, ConfigFactory}
+import com.wavesplatform.dex.api.http.entities.HttpRates
+import com.wavesplatform.dex.domain.asset.Asset.Waves
+import com.wavesplatform.dex.it.api.RawHttpChecks
+import com.wavesplatform.dex.it.docker.apiKey
+import com.wavesplatform.it.MatcherSuiteBase
+
+class UpdateRatesByAssetIdSpec extends MatcherSuiteBase with RawHttpChecks {
+
+  val defaultRates: HttpRates = Map(Waves -> 1d)
+
+  override protected def dexInitialSuiteConfig: Config = ConfigFactory.parseString(
+    s"""waves.dex {
+       |  price-assets = [ "$BtcId", "$UsdId", "WAVES" ]
+       |}""".stripMargin
+  )
+
+  override protected def beforeAll(): Unit = {
+    wavesNode1.start()
+    broadcastAndAwait(IssueBtcTx, IssueUsdTx)
+    dex1.start()
+  }
+
+  "PUT /matcher/settings/rates/{assetId} " - {
+
+    "should update rate by asset id " in {
+
+      withClue(" - asset doesn't have a rate") {
+        validate201Json(dex1.rawApi.upsertRate(usd, 0.01)).message should be(s"The rate 0.01 for the asset $UsdId added")
+      }
+
+      withClue(" - asset already have a rate") {
+        validate200Json(dex1.rawApi.upsertRate(usd, 0.02)).message should be(
+          s"The rate for the asset $UsdId updated, old value = 0.01, new value = 0.02"
+        )
+      }
+    }
+
+    "should return an error for incorrect rate values" in {
+      validateMatcherError(dex1.rawApi.upsertRate(btc, -1), StatusCodes.BadRequest, 20971535, "Asset rate should be positive")
+      validateMatcherError(dex1.rawApi.upsertRate(btc, 0), StatusCodes.BadRequest, 20971535, "Asset rate should be positive")
+    }
+
+    //TODO: now method return OK set rate to Infinite
+    "should return error if  the rate value more than Double.max" ignore {
+      validateMatcherError(dex1.rawApi.upsertRate(btc, Double.MaxValue + 1), StatusCodes.NotFound, 11534345, "Asset rate should be positive")
```

Comment:
If you want to send the bigger value, probably you should send a raw HTTP request. Probably we did something like this, but I can't remember. I created a task DEX-984 to make this possible through DexApi

Reply:
yep, I've done it
in this test JSON is correct, but the digit isn't